tests/admin_views/tests.py

```diff
@@ -1480,6 +1481,14 @@ def test_login(self):
         form = login.context[0].get('form')
         self.assertEqual(form.errors['username'][0], 'This field is required.')
 
+        # Test logon redirects when going directly to /admin/login (#25032)
+        self.client.get(reverse('admin:logout'))
+        response = self.client.get(reverse('admin:login'), follow=False)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, '<input type="hidden" name="next" value="%s" />' % reverse('admin:index'))
+        self.assertNotContains(response, '<input type="hidden" name="next" value="%s" />' % reverse('admin:login'))
+        self.client.get(reverse('admin:logout'))
```

Comment:
there's no need to "cleanup" by logging out as each test creates a new test client